vault/identity_store_util.go

```diff
@@ -925,6 +937,38 @@ func (i *IdentityStore) MemDBAliasByID(aliasID string, clone bool, groupAlias bo
 	return i.MemDBAliasByIDInTxn(txn, aliasID, clone, groupAlias)
 }
 
+func (i *IdentityStore) MemDBAliasByAlternativeIDInTxn(txn *memdb.Txn, alternativeID string, clone bool, groupAlias bool) (*identity.Alias, error) {
```

Comment:
A godoc here would be great, since this might be confusing otherwise. Perhaps something like:

```
// MemDBAliasByAlternativeIDInTxn finds an alias by one of its alternative IDs, which are IDs that can also be used to uniquely identify this alias.
```